exec/java-exec/src/main/java/org/apache/drill/exec/expr/ClassGenerator.java

```diff
@@ -646,7 +687,7 @@ public void preparePlainJava() {
         Class<?> p = params[i];
         childNew.arg(shim.param(model._ref(p), "arg" + i));
       }
-      shim.body()._return(childNew);
+      shim.body()._return(JExpr._this().invoke("injectMembers").arg(childNew));
```

Comment:
Could you please explain what is the reason for this change?

Reply:
The innerClasses like BatchHolder may have the constants, BatchHolder instance is created via newBatchHolder dynamically when processing recordBatch, that is the difference between innerClasses and nested splitting classes.

The innerClasses member fields is initialized by invoking injectMembers(innerClassInstance) which is override generated by this code as shown in 
`protected BatchHolder newBatchHolder(int batchRowCount) {
    return this.injectMembers(new BatchHolder(batchRowCount));
  }`